src/extension.ts

```diff
@@ -199,6 +202,69 @@ export async function activate(context: vscode.ExtensionContext, perfStats: { lo
     ext.managedIdentityBranchDataProvider = new ManagedIdentityBranchDataProvider();
     ext.v2.api.resources.registerAzureResourceBranchDataProvider(AzExtResourceType.ManagedIdentityUserAssignedIdentities, ext.managedIdentityBranchDataProvider);
 
+
+    const AUTH_PROVIDER_ID = 'microsoft'; // VS Code Azure auth provider
+    const SCOPES = ['https://management.azure.com/.default']; // Default ARM scope
+
+    registerEvent(
+        'treeView.onDidChangeSessions',
+        vscode.authentication.onDidChangeSessions,
+        async (context: IActionContext, _e: vscode.AuthenticationSessionsChangeEvent) => {
+            context.errorHandling.suppressDisplay = true;
+            context.telemetry.suppressIfSuccessful = true;
+            context.telemetry.properties.isActivationEvent = 'true';
+
+            try {
+                const accounts = await vscode.authentication.getAccounts(AUTH_PROVIDER_ID);
+                if (accounts.length === 0) {
+                    ext.outputChannel.appendLine('No Azure accounts found.');
+                    return;
+                }
+
+                const session = await vscode.authentication.getSession(
```

Comment:
You should pass options to this to make sure it is a silent auth attempt, or else the user could be prompted seemingly out of the blue

Reply:
added silent:true option